### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by rishabh-browserstack**

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+		t.Log(build, err)
+		require.Equal(t, "", build)
+		require.Error(t, err)
+	}
+
+}
+
+func TestCheckBuildStatus(t *testing.T) {
+	t.Log("It should throw if build_id is not passed")
```
Comment (deepwired) [inline main_test.go:60]: what error is supposed to be thrown here?

File: bitrise.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  # An example secret param, define it (A_SECRET_PARAM) in .bitrise.secrets.yml
+  - A_SECRET_PARAM: $A_SECRET_PARAM
+  # If you want to share this step into a StepLib
+  - BITRISE_STEP_ID: espresso-browserstack-official
+  - BITRISE_STEP_VERSION: "0.0.1"
+  - BITRISE_STEP_GIT_CLONE_URL: https://github.com/browserstack/browserstack-bitrise-espresso-step.git
+  - MY_STEPLIB_REPO_FORK_GIT_URL: $MY_STEPLIB_REPO_FORK_GIT_URL
+
+workflows:
+  test:
```
Comment (deepwired) [inline bitrise.yml:15]: we can remove test: right?

File: services.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	payload, err := json.Marshal(payload_values)
+
+	final_payload := appendExtraCapabilities(string(payload))
+
+	// log.Print("Final payload -> ", string(final_payload))
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", BROWSERSTACK_DOMAIN+APP_AUTOMATE_BUILD_ENDPOINT, bytes.NewBuffer(final_payload))
+
+	req.SetBasicAuth(username+"-bitrise", access_key)
```
Comment (deepwired) [inline services.go:34]: this works out of the box?

File: services.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	err := multipart_writer.Close()
+
+	if err != nil {
+		return "", errors.New(INVALID_FILE_TYPE_ERROR)
+	}
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", BROWSERSTACK_DOMAIN+endpoint, payload)
+
+	req.SetBasicAuth(username, access_key)
```
Comment (deepwired) [inline services.go:87]: do we have to append -bitrise for username?
if yes? can we constant it for re-usability.

File: services.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+		}
+
+		build_status = build_parsed_response["status"].(string)
+
+	}, POOLING_INTERVAL, false)
+
+	// infinite loop -> consider this as a ticker
+	for {
+
+		if build_status != "running" && build_status != "" {
```
Comment (deepwired) [inline services.go:164]: if build_status != "running" && **build_status_error** != "" {

File: step.yml [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+      description: "Network logs captures the network traffic, latency, HTTP requests and responses in the HAR (HTTP Archive) format. "
+      value_options: 
+        - true
+        - false
+      category: "Debug logs"
+  - device_logs: false
+    opts:
+      title: "Android logcat logs"
+      summary: "Enable logs generated by Android logcat"
+      description: Device logs are system logs specific to your application generated by Android logcat that help you debug any test crashes.
```
Comment (deepwired) [inline step.yml:147]: quotes are missing in description, seems to be present everywhere else

File: step.yml [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        New line key-value pairs of capabilities provided by BrowserStack.
+        
+        For example:
+        coverage=true
+        
+        geoLocation=CN
+        
+        [Learn about more capabilities.](https://www.browserstack.com/docs/app-automate/api-reference/espresso/builds#execute-a-build)
+      category: "Test Configuration"
+
```
Comment (deepwired) [inline step.yml:280]: remove extra lines,

can we have a linter added in this repo as well? we can keep it as a dedicated task if not this one.

File: structs.go [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	ClearAppData           bool        `json:"clearPackageData,omitempty"`
+	SingleRunnerInvocation bool        `json:"singleRunnerInvocation,omitempty"`
+	Class                  []string    `json:"class,omitempty"`
+	Package                []string    `json:"package,omitempty"`
+	Annotation             []string    `json:"annotation,omitempty"`
+	Size                   []string    `json:"size,omitempty"`
+	UseMockServer          bool        `json:"allowDeviceMockServer,omitempty"`
+	UseTestSharding        interface{} `json:"shards,omitempty"`
+
+	// non ui fields
```
Comment (deepwired) [inline structs.go:47]: why are these commented?

**Review by deepwired — CHANGES_REQUESTED** [review]
Small Comments

File: services.go (hunk 4 above)
Comment (deepwired) [reply services.go:87]: resolved,
this is not currently supported in our existing structure, nor is it required, 
bitrise mentioned via build api is sufficient

File: services.go (hunk 3 above)
Comment (deepwired) [reply services.go:34]: resolved

File: bitrise.yml (hunk 2 above)
Comment (deepwired) [reply bitrise.yml:15]: resolved,
this workflow is used to trigger the test for this plugin on bitrise with sample inputs/params

File: services.go (hunk 5 above)
Comment (deepwired) [reply services.go:164]: resolved, there are scenarios during initiation where build_status = ""

File: bitrise.yml (hunk 2 above)
Comment (rishabh-browserstack) [reply bitrise.yml:15]: Discussed with @deepwired on the call, marking as resolved

File: step.yml (hunk 7 above)
Comment (rishabh-browserstack) [reply step.yml]: removed extra lines and added a lint config

File: structs.go (hunk 8 above)
Comment (rishabh-browserstack) [reply structs.go]: As discussed with @deepwired over the call, added a better comment which explains why we've commented params.

**Review by deepwired — APPROVED** [review]
LGTM

**Pull request merged**